prog/vm/nexus.rb

```diff
@@ -240,6 +241,16 @@ def allocate
     vm_host_id
   end
 
+  def allocate_boot_image(vm_host, storage_volume, boot_image_name)
```

Comment:
This part conflicts with @bsatzger's allocator improvement: https://github.com/ubicloud/ubicloud/pull/1551. Whoever merges first, the other will need to incorporate these changes. fyi